wrappers/dlib/face/validate_face.cpp

```diff
@@ -0,0 +1,162 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2019 Intel Corporation. All Rights Reserved.
+
+#include "validate_face.h"
+#include "markup_68.h"
+
+
+/*
+    Calculates the average depth for a range of two-dimentional points in face, such that:
+        point(n) = face.part(n)
+    and puts the result in *p_average_depth.
+
+    Points for which no depth is available (is 0) are ignored and not factored into the average.
+
+    Returns true if an average is available (at least one point has depth); false otherwise.
+*/
+bool find_depth_from(
+    rs2::depth_frame const & frame,
+    float const depth_scale,
+    dlib::full_object_detection const & face,
+    int markup_from, int markup_to,
+    float * p_average_depth
```

Comment:
You could use & instead - then you don't need to check for nullptr

Reply:
Is there a convention? I personally prefer, for readability, to 'calc_something( &result );' where the & makes it clear that the result gets written to result.